test/unit/remote-config/remote-config.spec.ts

```diff
@@ -193,6 +221,143 @@ describe('RemoteConfig', () => {
+        response.versions[0].description = invalidDescription as any;
+        const stub = sinon
+          .stub(RemoteConfigApiClient.prototype, 'listVersions')
+          .resolves(response);
+        stubs.push(stub);
+        return remoteConfig.listVersions()
+          .should.eventually.be.rejected.and.have.property('message',
+            'Version description must be a non-empty string');
+      });
+    });
+
+    ['', 123, 1.2, null, NaN, true, [], {}].forEach((invalidRollbackSource) => {
+      it(`should reject if the rollbackSource is: ${invalidRollbackSource}`, () => {
+        const response = deepCopy(REMOTE_CONFIG_LIST_VERSIONS_RESULT);
+        response.versions[0].rollbackSource = invalidRollbackSource as any;
+        const stub = sinon
+          .stub(RemoteConfigApiClient.prototype, 'listVersions')
+          .resolves(response);
+        stubs.push(stub);
+        return remoteConfig.listVersions()
+          .should.eventually.be.rejected.and.have.property('message',
+            'Version rollback source must be a non-empty string');
+      });
+    });
+
+    ['', 'abc', 123, 1.2, null, NaN, [], {}].forEach((invalidIsLegacy) => {
+      it(`should reject if the isLegacy is: ${invalidIsLegacy}`, () => {
+        const response = deepCopy(REMOTE_CONFIG_LIST_VERSIONS_RESULT);
+        response.versions[0].isLegacy = invalidIsLegacy as any;
+        const stub = sinon
+          .stub(RemoteConfigApiClient.prototype, 'listVersions')
+          .resolves(response);
+        stubs.push(stub);
+        return remoteConfig.listVersions()
+          .should.eventually.be.rejected.and.have.property('message',
+            'Version.isLegacy must be a boolean');
+      });
+    });
+
+    ['', 'abc', 123, 1.2, null, NaN, [], {}].forEach((invalidUpdateTime) => {
+      it(`should reject if the updateTime is: ${invalidUpdateTime}`, () => {
+        const response = deepCopy(REMOTE_CONFIG_LIST_VERSIONS_RESULT);
+        response.versions[0].updateTime = invalidUpdateTime as any;
+        const stub = sinon
+          .stub(RemoteConfigApiClient.prototype, 'listVersions')
+          .resolves(response);
+        stubs.push(stub);
+        return remoteConfig.listVersions()
+          .should.eventually.be.rejected.and.have.property('message',
+            'Version update time must be a valid ISO date string');
+      });
+    });
+
+    it('should resolve with template versions list on success', () => {
+      const stub = sinon
+        .stub(RemoteConfigApiClient.prototype, 'listVersions')
+        .resolves(REMOTE_CONFIG_LIST_VERSIONS_RESULT);
+      stubs.push(stub);
+      return remoteConfig.listVersions({
+        pageSize: 2
+      })
+        .then((response) => {
+          expect(response.versions.length).to.equal(2);
+          expect(response.versions[0].updateTime).equals('Thu, 07 May 2020 18:46:09 GMT');
+          expect(response.versions[1].updateTime).equals('Thu, 07 May 2020 18:44:41 GMT');
+        });
+    });
```

Comment:
Add test cases for input timestamp conversion (Date to ISO and UTC to ISO) and version number conversion (75 to '75').

Reply:
Input timestamp conversion tests are here https://github.com/firebase/firebase-admin-node/pull/896/files#diff-e21807cc3d787a27c233f0b4c5cfc719R442
Do you think we need specific ones for each timestamp?

Also, the API responds to both number and string version numbers... do you think we should always convert them to string?

Reply:
Ah. It's a little buried in the code. Can we make it a bit more explicit?